libnetwork/firewall_linux.go

```diff
@@ -21,24 +21,35 @@ func setupArrangeUserFilterRule(c *controller) {
 // IPTableForwarding is disabled, because it contains rules configured by user that
 // are beyond docker engine's control.
 func arrangeUserFilterRule() {
-	if ctrl == nil || !ctrl.iptablesEnabled() {
-		return
-	}
-	// TODO IPv6 support
-	iptable := iptables.GetIptable(iptables.IPv4)
-	_, err := iptable.NewChain(userChain, iptables.Filter, false)
-	if err != nil {
-		logrus.Warnf("Failed to create %s chain: %v", userChain, err)
+	if ctrl == nil {
 		return
 	}
 
-	if err = iptable.AddReturnRule(userChain); err != nil {
-		logrus.Warnf("Failed to add the RETURN rule for %s: %v", userChain, err)
-		return
+	conds := map[iptables.IPVersion]bool{
+		iptables.IPv4: ctrl.iptablesEnabled(),
+		iptables.IPv6: ctrl.ip6tablesEnabled(),
 	}
```

Comment:
Troubleshooting ibnetwork is hard enough without having to deal with firewall rules being set in a nondeterministic order.
```suggestion
	conds := []struct{ ipVer iptables.IPVersion, cond bool }{
		{ iptables.IPv4, ctrl.iptablesEnabled() },
		{ iptables.IPv6, ctrl.ip6tablesEnabled() },
	}
```